database creation

anishaf92 opened the pull request.

vigneshsudhakar reviewed (changes requested):
_pycache_/
venv/
.idea/

please add these 3 to the .gitignore file and remove .idea folder which you have pushed to this branch

vigneshsudhakar commented on models.py:
```diff
@@ -0,0 +1,98 @@
+    return engine
+
+
+cart = db.Table('cart',
```
Why is this different from the others?
Please maintain uniformity.

vigneshsudhakar commented on models.py:
```diff
@@ -0,0 +1,98 @@
+
+
+class User(db.Model):
+    __tablename__ = "user"
```
" __tablename__ = "user""

Why is this required in every class?

vigneshsudhakar commented on models.py:
```diff
@@ -0,0 +1,98 @@
+    username = db.Column(db.String(80), unique=True, nullable=False)
+    email = db.Column(db.String(120), unique=True, nullable=False)
+    mobile = db.Column(db.String(200), unique=True, nullable=False)
+    items_in_cart = db.relationship('Product', secondary=cart, backref='cart-products')
```
Please add "to_json" function to each class and return a json object containing all the data

vigneshsudhakar replied:
is_active = db.Column(db.Boolean, default=True, nullable=False)
is_deleted = db.Column(db.Boolean, default=False, nullable=False)
created_by = db.Column(db.Integer, nullable=True)
created_date = db.Column(db.DateTime, default=datetime.datetime.utcnow, nullable=False)
updated_by = db.Column(db.Integer, nullable=True)
updated_date = db.Column(db.DateTime, default=datetime.datetime.utcnow, nullable=False)



Add these above 6 fields to all the tables

vigneshsudhakar commented on requirements.txt:
```diff
@@ -0,0 +1,13 @@
+click==8.1.3
```
Why are all these packages added here?
Are they needed?
I see you using just Flask, mysql-connector, dotenv and SQLAlchemy

anishaf92 commented:
> _pycache_/ venv/ .idea/
> 
> please add these 3 to the .gitignore file and remove .idea folder which you have pushed to this branch

anishaf92 commented:
Removed .idea folder and added pycache/ venv/ .idea/ files to gitgnore

> _pycache_/ venv/ .idea/
> 
> please add these 3 to the .gitignore file and remove .idea folder which you have pushed to this branch

Removed .idea folder and added pycache/ venv/ .idea/ files to gitgnore
